### PR TITLE
Corrected autolinking urls without protocol

### LINKED ISSUE #11040: Auto-link correction
## 📝 Provide detailed reproduction steps (if any)

1. …type www.google.com in content (without http)
2. …inspect the link, you'll find it with href='www.google.com' which is invalid

### ✔️ Expected result

href='https://www.google.com'

### ❌ Actual result

href='www.google.com'

### ❓ Possible solution

Either auto-link would add https:// to the link, or doesn't link www.google.com

### PR DESCRIPTION
**Pull request by kbinieda**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Fix (link): Corrected autolinking URLs without protocol. Closes #11040.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

File: packages/ckeditor5-link/src/utils.js [hunk 1]
```diff
@@ -165,15 +165,26 @@ export function isEmail( value ) {
...
 
 	return link && isProtocolNeeded ? protocol + link : link;
 }
 
+/**
+ * Checks if protocol is already included in the link
+ *
+ *
+ * @param {String} link
+ * @returns {boolean}
```
Comment (scofalik) [inline packages/ckeditor5-link/src/utils.js:182]:
```suggestion
 * @returns {Boolean}
```

File: packages/ckeditor5-link/src/autolink.js [hunk 2]
```diff
@@ -233,16 +233,16 @@ export default class AutoLink extends Plugin {
...
+		if ( !this.isEnabled || !isLinkAllowedOnRange( range, model ) || !linkHasProtocol( parsedUrl ) ) {
 			return;
 		}
 
 		// Enqueue change to make undo step.
 		model.enqueueChange( writer => {
-			const defaultProtocol = this.editor.config.get( 'link.defaultProtocol' );
-			const parsedUrl = addLinkProtocolIfApplicable( link, defaultProtocol );
 			writer.setAttribute( 'linkHref', parsedUrl, range );
-
```
Comment (scofalik) [inline packages/ckeditor5-link/src/autolink.js:245]: Please don't remove existing empty lines :) (unless this was automated by IDE, btw. @DawidKossowski had the same thing).

Personally, I am a big fan of empty lines and separating code blocks. I think that in our codebase most of use like adding extra lines for readability reasons.

File: packages/ckeditor5-link/tests/autolink.js [hunk 3]
```diff
@@ -237,6 +237,15 @@ describe( 'AutoLink', () => {
 			);
 		} );
 
+		it( 'not autolinks address without protocol if defaultProtocol is not valid', () => {
```
Comment (scofalik) [inline packages/ckeditor5-link/tests/autolink.js:240]:
```suggestion
		it( 'does not autolink address without protocol (defaultProtocol is not set or not valid)', () => {
```

**Review by scofalik — CHANGES_REQUESTED** [review]
Very minor things, as commented.

File: packages/ckeditor5-link/src/autolink.js (hunk 2 above)
Comment (kbinieda) [reply packages/ckeditor5-link/src/autolink.js:245]: Right, I am also a fan of separating code blocks, but in this case I've removed most of the code from the block so it made sense to me to merge both code blocks together. On the other hand it makes sense to visually separate actual command changes from broadcasting another, undo command.

File: packages/ckeditor5-link/src/utils.js [hunk 4]
```diff
@@ -165,15 +165,26 @@ export function isEmail( value ) {
...
-	const isProtocolNeeded = !!protocol && !PROTOCOL_REG_EXP.test( link );
+	const isProtocolNeeded = !!protocol && !linkHasProtocol( link );
 
 	return link && isProtocolNeeded ? protocol + link : link;
 }
 
+/**
+ * Checks if protocol is already included in the link
+ *
+ *
```
Comment (scofalik) [inline packages/ckeditor5-link/src/utils.js:180]:
```suggestion
 * Checks if protocol is already included in the link.
 *
```

**Pull request merged**